### ISSUE TITLE
Hero Core ERR_INVALID_URL

### ISSUE THREAD

**Issue by kriskratz** [ulixee/hero#321]
Hero is crashing when loading a [webpage](https://mysocialpractice.com/2024/08/is-dental-seo-dead/) that contains an HTML Script that includes a malformed url: `https:// /vfRg2L1ej33bleY00JdN9pxuSvOX2mNI-nTw9UpUoPg`

Is there a way I can handle this error so things keep going?

HTML snippet: 
```html
<script type="rocketlazyloadscript" data-rocket-type='text/javascript'>
(function(f,b,n,j,x,e){x=b.createElement(n);e=b.getElementsByTagName(n)[0];x.async=1;x.src=j;e.parentNode.insertBefore(x,e);})(window,document,'script','https:// /vfRg2L1ej33bleY00JdN9pxuSvOX2mNI-nTw9UpUoPg');
</script>
``` 

Complete Error: 
```shell
2024-10-18T05:49:27.520Z ERROR [hero-core/index] UnhandledError(fatal) {
  origin: 'uncaughtException',
  context: {},
  sessionId: null,
  sessionName: undefined
} TypeError: Invalid URL
    at new URL (node:internal/url:796:36)
    at WebsocketSession.isWebsocketUrl (/...localpath.../agent/main/lib/WebsocketSession.ts:65:20)
    at NetworkManager.onNetworkRequestWillBeSent (/...localpath.../agent/main/lib/NetworkManager.ts:311:31)
    at DevtoolsSession.emit (node:events:518:28)
    at DevtoolsSession.emit (node:domain:488:12)
    at DevtoolsSession.emit (/...localpath.../node_modules/commons/lib/TypedEventEmitter.ts:162:18)
    at DevtoolsSession.onMessage (/...localpath.../agent/main/lib/DevtoolsSession.ts:117:12)
    at Connection.onMessage (/...localpath.../agent/main/lib/Connection.ts:87:23)
    at process.processImmediate (node:internal/timers:480:21) {
  code: 'ERR_INVALID_URL',
  input: 'https://%20/vfRg2L1ej33bleY00JdN9pxuSvOX2mNI-nTw9UpUoPg'
}
TypeError: Invalid URL
    at new URL (node:internal/url:796:36)
    at WebsocketSession.isWebsocketUrl (/...localpath.../agent/main/lib/WebsocketSession.ts:65:20)
    at NetworkManager.onNetworkRequestWillBeSent (/...localpath.../agent/main/lib/NetworkManager.ts:311:31)
    at DevtoolsSession.emit (node:events:518:28)
    at DevtoolsSession.emit (node:domain:488:12)
    at DevtoolsSession.emit (/...localpath.../node_modules/commons/lib/TypedEventEmitter.ts:162:18)
    at DevtoolsSession.onMessage (/...localpath.../agent/main/lib/DevtoolsSession.ts:117:12)
    at Connection.onMessage (/...localpath.../agent/main/lib/Connection.ts:87:23)
    at process.processImmediate (node:internal/timers:480:21) {
  code: 'ERR_INVALID_URL',
  input: 'https://%20/vfRg2L1ej33bleY00JdN9pxuSvOX2mNI-nTw9UpUoPg'
}
```

**Comment by kriskratz** [comment]
I modified `@ulixee/unblocked-agent/lib/WebsocketSession.js` and it seems to have fixed the problem for me.

I added a URL parse check to isWebsocketUrl(), `if(!URL.canParse(url)) return false;` 

Complete code block:
```typescript
isWebsocketUrl(url) {
        if(!URL.canParse(url)) return false;
        const parsed = new URL(url);
        return (parsed.hostname === this.host &&
            parsed.port === this.port.toString() &&
            parsed.searchParams.get('secret') === this.secret);
    }
```

**Comment by blakebyrnes** [comment]
Thanks! I think this got fixed in #320 but hasn't been put into a published npm version yet
